defrag/modules/search.py

```diff
@@ -0,0 +1,37 @@
+from defrag.modules.helpers import Query, QueryResponse
+from typing import List
+from pydantic import BaseModel
+from defrag import app
+from defrag.modules.helpers.services_manager import ServicesManager
+from defrag.modules.bugs import search as bugs_search
+from defrag.modules.docs import search_merging_sources_docs as docs_search
+from defrag.modules.twitter import search_twitter as twitter_search
+from defrag.modules.reddit import search_reddit as reddit_search
+from defrag.modules.wikis import search as wikis_search
```

Comment:
I think it would be nicer to have a search function with always the same name in every module. Then, instead of importing all of them here, we can just loop over all the modules that support search and call their respective search function with our keyword. This would allow us to add new modules without needing to change this file.

Reply:
Look at what I do with OpenGM for help strings:
When importing the modules, I check if they have a help function:
```python
if hasattr(imported_module, "__help__") and imported_module.__help__:
        HELPABLE[imported_module.__mod_name__.lower()] = imported_module
```
Then, when the help text is requested, it calls `__help__` on every module that is in `HELPABLE` and makes a nice output out of all the different textes.

Reply:
hehe look at what I do in __main__ with register() 👯

Reply:
More seriously, what you're saying makes a lot of sense, but for now I guess it's preferrable to do things bottom-up when we can, that makes it easier to debug and add/remove things from scope.

Reply:
Okay will do

Reply:
> hehe look at what I do in __main__ with register() 👯 

Yeah, similar to that, but not all modules might be searchable, so we need to keep that in mind

Reply:
see https://github.com/openSUSE/defrag-api/pull/63/commits/195e71aa682081f47543fa198451f2e42a4f5477